db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/SYSSCHEMASRowFactory.java

```diff
@@ -271,4 +271,13 @@ public TupleDescriptor buildDescriptor(
 	public static final String SYSSCHEMASVIEW_VIEW_SQL1 = "create view sysschemasView as \n" +
 			SUPER_USER_SCHEMA;
 
+
+	public static final String RANGER_USER_SCHEMA =
+            "select S.* from SYS.SYSSCHEMAS as S where S.SCHEMANAME not in " +
```

Comment:
Why do we choose double-negation here to get the accessible schemas? That is, NOT IN + SchemaFilterVTI returning the list of not-accessible schemas. Why don't we use the combination of IN + SchemaFilterVTI returning the list of accessible schemas? The latter usually is more performant.

Reply:
The reason is when using `in`, it will optimized to `join`. And in the join operation, `RangerAuthorizer.canSeeSchema` will through null pointer error from `lcc`. I don't know why join operation will have error but using `not in` to avoid join can avoid this situation.

Reply:
I see @wb14123. I'm a little concerned about the null pointer exception for join. Maybe we should understand better this exception and see if the same problem could surface when the whole ranger view(SYSSCHEMASVIEW_VIEW_RANGER) is used as the right side of a join. Thanks.